src/teacher.py

```diff
@@ -60,9 +60,13 @@ def _answer_Demonstration_query(
         except (ApproachTimeout, ApproachFailure):
             return DemonstrationResponse(query, teacher_traj=None)
 
-        traj, _, goal_reached = utils.run_policy_on_task(
```

Comment:
can we just purge `run_policy_on_task` everywhere? where is it even used now? isn't it just a super light wrapper around run_policy_until?

Reply:
i looked through the code, it seems like the things that use `run_policy_on_task` are:
* demo creation
* main.py for evaluation and video creation

I feel like we should just use `run_policy_until` in both those places. All these utility methods are getting too confusing! Would be nice to just have `run_policy_until`, `simulate_policy_until`, and `option_to_trajectory` (the latter two would not be used in the final system).

(Of course we'd have to keep `run_policy_on_task` in tests/utils.py)

Reply:
after reading below, i also feel like we should just wipe out `option_to_trajectory` too

Reply:
works for me!